components/gateway/pkg/plugins/audit.go

```diff
@@ -0,0 +1,394 @@
+					return nil, h.Errf("failed to parse publisher_kafka_enabled: %v", err)
+				}
+
+			case "publisher_kafka_broker":
+				if !h.AllArgs(&a.PublisherKafkaBroker) {
+					return nil, h.Errf("expected one string value for kafka_broker")
+				}
+
+			case "publisher_kafka_tls_enabled":
+				var err error
+				a.PublisherKafkaTLSEnabled, err = parseBool(h.Dispenser)
+				if err != nil {
+					return nil, h.Errf("failed to parse publisher_kafka_tls_enabled: %v", err)
+				}
+
+			case "publisher_kafka_sasl_enabled":
+				var err error
+				a.PublisherKafkaSASLEnabled, err = parseBool(h.Dispenser)
+				if err != nil {
+					return nil, h.Errf("failed to parse publisher_kafka_sasl_enabled: %v", err)
+				}
+
+			case "publisher_kafka_sasl_username":
+				if !h.AllArgs(&a.PublisherKafkaSASLUsername) {
+					return nil, h.Errf("expected one string value for publisher kafka sasl username")
+				}
+
+			case "publisher_kafka_sasl_password":
+				if !h.AllArgs(&a.PublisherKafkaSASLPassword) {
+					return nil, h.Errf("expected one string value for publisher kafka sasl password")
+				}
+
+			case "publisher_kafka_sasl_mechanism":
+				if !h.AllArgs(&a.PublisherKafkaSASLMechanism) {
+					return nil, h.Errf("expected one string value for publisher kafka sasl mechanism")
+				}
+
+			case "publisher_kafka_sasl_scram_sha_size":
+				var publisherKafkaSaslScramShaSize string
+				if !h.AllArgs(&publisherKafkaSaslScramShaSize) {
+					return nil, h.Errf("expected one boolean value")
+				}
+
+				res, err := strconv.ParseInt(publisherKafkaSaslScramShaSize, 10, 32)
+				if err != nil {
+					return nil, h.Errf("failed to parse publisher_kafka_sasl_scram_sha_size: %v", err)
+				}
+				a.PublisherKafkaSASLScramSHASize = int(res)
+
+			case "publisher_nats_enabled":
+				var err error
+				a.PublisherNatsEnabled, err = parseBool(h.Dispenser)
+				if err != nil {
+					return nil, h.Errf("failed to parse publisher_nats_enabled: %v", err)
+				}
+
+			case "publisher_nats_url":
+				if !h.AllArgs(&a.PublisherNatsURL) {
+					return nil, h.Errf("expected one string value for publisher_nats_url")
+				}
+
+			case "publisher_nats_client_id":
+				if !h.AllArgs(&a.PublisherNatsClientId) {
+					return nil, h.Errf("expected one string value for publisher_nats_client_id")
+				}
+
+			default:
+				return nil, h.Errf("unrecognized option: %s", key)
+			}
+		}
+	}
+
+	return a, nil
+}
+
+func parseBool(d *caddyfile.Dispenser) (bool, error) {
+	var b string
+	if !d.AllArgs(&b) {
+		return false, d.Errf("expected one boolean value")
+	}
+
+	res, err := strconv.ParseBool(b)
+	if err != nil {
+		return false, d.Errf("expected boolean value")
+	}
+
+	return res, nil
+}
+
+// Implements the caddy.Provisioner interface.
+func (a *Audit) Provision(ctx caddy.Context) error {
+	// We have to instantiate a different logger here instead of using
+	// caddy's logger because we need a different interface for the watermill
+	// logger.
+	a.logger = logging.NewLogrus(logrus.New())
```

Comment:
Can you implement the watermil logger interface using the caddy logger as underlying logger?